node/router/src/inbound.rs

```diff
@@ -202,15 +201,13 @@ pub trait Inbound<N: Network>: Reading + Outbound<N> {
                 // Decrement the number of puzzle requests.
                 self.router().cache.decrement_outbound_puzzle_requests(peer_ip);
 
-                // Clone the serialized message.
-                let serialized = message.clone();
                 // Perform the deferred non-blocking deserialization of the block header.
                 let header = match message.block_header.deserialize().await {
                     Ok(header) => header,
                     Err(error) => bail!("[PuzzleResponse] {error}"),
                 };
                 // Process the puzzle response.
```

Comment:
Perhaps this doc needs some updating?

Reply:
Actually, it should still be ok, as the arguments consist of the contents of that message type.